pyo3-macros-backend/src/method.rs

```diff
@@ -351,52 +332,70 @@ impl<'a> FnSpec<'a> {
                 .map(|stripped| syn::Ident::new(stripped, name.span()))
         };
 
-        let (fn_type, skip_first_arg, fixed_convention) = match fn_type_attr {
-            Some(MethodTypeAttribute::StaticMethod) => (FnType::FnStatic, false, None),
-            Some(MethodTypeAttribute::ClassAttribute) => (FnType::ClassAttribute, false, None),
-            Some(MethodTypeAttribute::New) | Some(MethodTypeAttribute::NewClassMethod) => {
+        let (fn_type, skip_first_arg, fixed_convention) = match method_attributes.as_mut_slice() {
```

Comment:
IMO `skip_first_arg` should really be a boolean-like enum, there are too many trues and falses in here.

Reply:
I agree but I'll deal with that in a follow-up PR.

Reply:
Actually I pushed a second commit to address this, which moves this to be a method on the `FnType` enum.